port/webhook/refreshWebhookState.go

```diff
@@ -0,0 +1,67 @@
+package webhook
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/port-labs/terraform-provider-port-labs/internal/cli"
+	"github.com/port-labs/terraform-provider-port-labs/internal/flex"
+)
+
+func refreshWebhookState(ctx context.Context, state *WebhookModel, w *cli.Webhook) error {
+	state.ID = types.StringValue(w.Identifier)
+	state.Identifier = types.StringValue(w.Identifier)
+	state.CreatedAt = types.StringValue(w.CreatedAt.String())
+	state.CreatedBy = types.StringValue(w.CreatedBy)
+	state.UpdatedAt = types.StringValue(w.UpdatedAt.String())
+	state.UpdatedBy = types.StringValue(w.UpdatedBy)
+	state.Icon = flex.GoStringToFramework(w.Icon)
+	state.Title = flex.GoStringToFramework(w.Title)
+	state.Description = flex.GoStringToFramework(w.Description)
+	state.Enabled = flex.GoBoolToFramework(w.Enabled)
+
+	if w.Security.RequestIdentifierPath != nil || w.Security.Secret != nil || w.Security.SignatureHeaderName != nil || w.Security.SignatureAlgorithm != nil || w.Security.SignaturePrefix != nil {
+		state.Security = &SecurityModel{
+			Secret:                flex.GoStringToFramework(w.Security.Secret),
+			SignatureHeaderName:   flex.GoStringToFramework(w.Security.SignatureHeaderName),
+			SignatureAlgorithm:    flex.GoStringToFramework(w.Security.SignatureAlgorithm),
+			SignaturePrefix:       flex.GoStringToFramework(w.Security.SignaturePrefix),
+			RequestIdentifierPath: flex.GoStringToFramework(w.Security.RequestIdentifierPath),
+		}
+	}
+
+	if len(w.Mappings) > 0 {
```

Comment:
consider converting to some sort of array.prototype.map() equivilant (such ash funk.Map)